app/controllers/application_controller.rb

```diff
@@ -18,12 +20,27 @@ def sign_in(login_userinfo)
     user = User.user_from_userinfo(login_userinfo)
 
     @current_user = user
+    renew_session
     session[:userinfo] = login_userinfo
   end
 
   def sign_out
     @current_user = nil
     session.delete(:userinfo)
+    session.delete(:session_timeout_at)
+  end
+
+  def renew_session
+    session[:session_timeout_at] = Time.current + ENV.fetch("SESSION_TIMEOUT_IN_MINUTES", 15).to_i.minutes
+  end
+
+  def check_session_expiration
+    if session[:session_timeout_at].present? && session[:session_timeout_at] < Time.current
```

Comment:
if `session[:session_timeout_at].nil?` then shouldn't it also kick the user out? or under what condition would that not be present and is that ok to renew the session if it's not?